### PR TITLE
working custom liger kernel for weighting cross entropy

### PR DESCRIPTION
**Pull request by amazingvince**

### CODE REVIEW

File: run_clm.py [hunk 1]
```diff
@@ -539,47 +538,66 @@ def tokenize_function(examples):
...
         if "train" not in tokenized_datasets:
             raise ValueError("--do_train requires a train dataset")
-        train_dataset = tokenized_datasets["train"]
+        # Skip first 1000 samples to use for eval
+        train_dataset = tokenized_datasets["train"].skip(1000)
         if data_args.max_train_samples is not None:
             max_train_samples = min(len(train_dataset), data_args.max_train_samples)
             train_dataset = train_dataset.select(range(max_train_samples))
 
+    apply_liger_kernel_to_chess_llama(model=model)
```
Comment (amazingvince) [inline run_clm.py:579]: you can choose what is monkey patched here. So we can turn off or change the loss function here or turn off one of the other triton kernels.

**Review by kfallah — APPROVED** [review]
Nice! Will this be a drop-in with the llama-chess PR (where we just apply the monkey patch to the new llama class)?
